### ISSUE TITLE
Character Pricing & Smarter Prices

### ISSUE THREAD

**Issue by fluffyCalculator** [currency-cop/currency-cop#69]
Doesn't look like you have a feature request place/tag so I made my own :P, so anyways. Would it be possible to have pricing of characters items, and pricing somewhat accurately priced based on mods. Like what poe trade macro has ( i think it goes through an api to get checked, like poeprices or something) Although it is not completely accurate, I still use it, you could make it a toggle if you wish too

**Comment by nijikokun** [comment]
Idea has been tossed around. That API is _very slow_ for a large number of items, order of minutes to hours depending on tabs. It's not scaleable. I wouldn't mind an experimental PR.

Character tabs is in a PR, i'm not sure about the demand, and it is also a very situational thing, when to poll, how to include it, especially across multiple portfolios.

**Comment by fluffyCalculator** [comment]
Right, yeah I think it would be slow, but what if you could set the tab that you want to price? Say a sell tab, trade macro takes a few seconds per item and poeprices takes a minute for a full quad tab of items, which isn't bad. 

Also I have noticed that whenever you move around tabs in game, the currency cop starts displaying the new tab in the same location as the old tab was (if that make sense), it doesn't recache the new tab's index.

**Comment by nijikokun** [comment]
It takes up to 10 minutes for a normal tab full of items, and up to 15 minutes for a quad tab since each item is priced individually. Also, the API doesn't give the amount of information required to do a properly display, I've talked a bit with the developer about this. 

Could you open a separate issue for the tab indexes? Thank you

**Comment by ajs** [comment]
Even without pricing based on stat-based lookup, one of the major issues appears to be that items are being priced based on legacy variants and other strange outliers. If you're not already, I strongly recommend using medians rather than means for calculating prices. This helps to eliminate some of the influence of outliers (e.g. if the prices are 1c, 7c, 20c, 3000c and 4000c, then 20c is the median, but the average would have been 1046c, which is a pretty huge difference!)

Ultimately, what you need is a pricing database with error bars, so that you can say that the price of an item is x+/-y, such as "Rumi's is worth 1c+/-100c" which makes it immediately obvious that there's some kind of major skew in Rumi's pricing (which is because of legacy variants). Right now, we just see that every Rumi's in our std inventory is worth 3,200c, which isn't very helpful.
